plans/software_tools.markdown

```diff
@@ -0,0 +1,103 @@
+We need to develop a coherent, effective package of recommended tools and settings for students in the program.
+
+# Critical Challenges Planning
+
+## The Challenge
+
+* What Critical Challenge are you going to work on?
+
+We need to develop a coherent, effective package of recommended tools and settings for students in the program.
+
+* Why is this a “challenge”?
+
+Getting all of the students' comptuers at a baseline so they can start learning in earnest is difficult because
+when students bring their computers, it is an unknown state.
+
+* Can it be solved? What would be value to the organization/team in solving it or working towards a solution?
+
+It can be solved, there are a number of tools that can help us in this regard.
+
+* What happens if we fail to solve it / improve it?
+
+We expend a large amount of person-hours getting students set up.
+
+* What skills do you have that make you the right person to work on this challenge?
+
+I have an ops background and know of some of the tools that can help fix this.
+
+## The Past
+
+* What’s been done so far?
+
+We have a tutorial on how to set up your environment.
+
+* Was it effective? How do you know?
+
+It works, people can make apps.
+
+* Was this someone’s challenge before you? Why did they stop doing it or how did we make progress without addressing it?
+
+Jeff and Josh spearheaded this. We've just thrown time at it.
+
+## The Present
+
+* What are the first steps that you’re ready to execute now?
+
+Start investigating tools like Vagrant.
+
+* What will be the next three things you do?
+
+Take a second look at the tutorial and look to improve it. Start building a vagrant server to test feasibility. See what the learning 
+curve is with a Linux VM.
+
+* How will you keep track of progress and results?
+
+Measure how many person-hours are used in setup and post setup troubleshooting.
+
+* Are you empowered to take the actions you think necessary?
+
+Yes.
+
+## The Future
+
+* What would total success look like?
+
+If we use less time to set up students computers.
+
+* What metrics can be used to evaluate results?
+
+How many class hours and outside of class hours are devoted to setup.
+
+* What would meaningful progress look like in 3 months? 6 months? 12 months?
+
+More instrcutional hours that can be devoted to curriculum instead of setup.
+
+* How feasible are each of those goals?
+
+On paper, pretty feasible.
+
+## The Periphery
+
+* What challenges are you going to stop working on to put time into this?
+
+This is probably the lowest priority challenge - it has become less of an issue since Macs have started to ship with ruby 2.0.0
+installed.
+
+* What research/learning do you need to do to be effective at solving it?
+
+Learn Varagnt!
+
+* What supports do you need to make meaningful progress?
+
+Maybe some hardware if this becomes successful.
+
+* Which other challenges does this align with?
+
+The onramping.
```

Comment:
Vagrant will give us the ability to allow anyone with a reasonably new computer run ruby, be it a PC or a Mac. They could get work done on any machine they had, and if they get a computer through Turing, it should theoretically be trivial to get the same environment set up.